src/Form/Transformer/RestoreRolesTransformer.php

```diff
@@ -17,38 +17,41 @@
 use Symfony\Component\Form\DataTransformerInterface;
 
 /**
- * @phpstan-implements DataTransformerInterface<array, array>
+ * @phpstan-implements DataTransformerInterface<string[], string[]>
  */
 class RestoreRolesTransformer implements DataTransformerInterface
 {
     /**
-     * @var array|null
+     * @var EditableRolesBuilder
      */
-    protected $originalRoles = null;
+    protected $rolesBuilder;
 
     /**
-     * @var EditableRolesBuilder|null
+     * @var string[]|null
      */
-    protected $rolesBuilder = null;
+    protected $originalRoles = null;
 
     public function __construct(EditableRolesBuilder $rolesBuilder)
     {
         $this->rolesBuilder = $rolesBuilder;
     }
 
+    /**
+     * @param string[]|null $originalRoles
+     */
     public function setOriginalRoles(?array $originalRoles = null): void
```

Comment:
Shouldn't we restrict the param to array ?

Reply:
I will be doing more changes on another Prs, I would say to focus this one on needed changes to comply with phpstan and psalm.